narwhals/_arrow/dataframe.py

```diff
@@ -174,13 +179,22 @@ def __getitem__(
                     )
                 msg = f"Expected slice of integers or strings, got: {type(item[1])}"  # pragma: no cover
                 raise TypeError(msg)  # pragma: no cover
-
             from narwhals._arrow.series import ArrowSeries
 
             # PyArrow columns are always strings
             col_name = item[1] if isinstance(item[1], str) else self.columns[item[1]]
+            assert not isinstance(item[0], str)  # help mypy  # noqa: S101
```

Comment:
Fancy changing this from `assert` to a `raise` statement as a follow up?

[the dangers of assert in python](https://snyk.io/blog/the-dangers-of-assert-in-python/)

TL;DR: `assert` is only run in debug mode, if you run python with the optimized flag, that will be skipped.

Reply:
Thanks for your comment @FBruzzesi , I replaced `assert` with `raise` in this PR https://github.com/narwhals-dev/narwhals/pull/980

Please, note that the line never gets hit so `# pragma: no cover` is added

Reply:
Oh I see, thanks for pointing that out 👌! Let's move the conversation there anyway